www/docs/guides/walletAccount.md

```diff
@@ -0,0 +1,185 @@
+- `networkChanged` : Triggered each time you change the current network in the wallet.
+
+At each change of network, both account and network events are occurring.  
+At each change of account, only the account event is occurring.
+
+### Subscribe
+
+#### accountsChanged
+
+```typescript
+const handleAccount: AccountChangeEventHandler = (accounts: string[] | undefined) => {
+  if (accounts?.length) {
+    const textAddr = accounts[0]; // hex string
+    setChangedAccount(textAddr); // from a React useState
+  }
+};
+selectedWalletSWO.on('accountsChanged', handleAccount);
+```
+
+#### networkChanged
+
+```typescript
+const handleNetwork: NetworkChangeEventHandler = (chainId?: string, accounts?: string[]) => {
+  if (!!chainId) {
+    setChangedNetwork(chainId);
+  } // from a React useState
+};
+selectedWalletSWO.on('networkChanged', handleNetwork);
+```
+
+### Un-subscribe :
+
+Similar to subscription, using `.off` method.
+
+```typescript
+selectedWalletSWO.off('accountsChanged', handleAccount);
+selectedWalletSWO.off('networkChanged', handleNetwork);
+```
+
+:::info
+You can subscribe both with the SWO or with a WalletAccount instance.  
+The above examples are using the SWO, because it's the simpler way to process.
+:::
+
+## Direct access to the wallet API entry points
+
+The WalletAccount class is able to interact with all the entrypoints of the Starknet wallet API, including some functionalities that do not exists in an Account class.
+You have a full description of this API [**here**](https://github.com/PhilippeR26/Starknet-WalletAccount/blob/main/doc/walletAPIspec.md).
+
+Some examples:
+
+### Request a change of wallet network
+
+Using your WalletAccount, you can ask to the wallet to change its current network:
+
+```typescript
+useEffect(
+  () => {
+    if (!isValidNetwork()) {
+      const tryChangeNetwork = async () => {
+        await myWalletAccount.switchStarknetChain(constants.StarknetChainId.SN_SEPOLIA);
+      };
+      tryChangeNetwork().catch(console.error);
+    }
+  },
+  [chainId] // from a networkChanged event
+);
+```
+
+![](./pictures/switchNetwork.png)
+
+### Request to display a token in the wallet
+
+Using your WalletAccount, you can ask to the wallet to display a new token:
+
+```typescript
+useEffect(
+  () => {
+    const fetchAddToken = async () => {
+      const resp = await myWalletAccount.watchAsset({
+        type: 'ERC20',
+        options: {
+          address: erc20Address,
+        },
+      });
+    };
+    if (isAirdropSuccess) {
+      fetchAddToken().catch(console.error);
+    }
+  },
+  [isAirdropSuccess] // from a React useState
+);
+```
+
+![](./pictures/addToken.png)
+
+## Change of network or account
+
+When you change the network or the account address, the WalletAccount is automatically updated, but it can lead to a tricky behavior (read and write in different networks, problems of Cairo versions of the accounts, ....).  
```

Comment:
```suggestion
When you change the network or the account address, the WalletAccount is automatically updated, but it can lead to tricky behavior (read and write in different networks, problems of Cairo versions of the accounts, ....).  
```

Reply:
Ok.